interoperator/pkg/internal/resources/resources.go

```diff
@@ -0,0 +1,481 @@
+		err := client.Get(context.TODO(), types.NamespacedName{
+			Name:      bindingID,
+			Namespace: namespace,
+		}, binding)
+		if err != nil {
+			log.Printf("error getting service binding. %v\n", err)
+			return nil, nil, nil, nil, err
+		}
+	}
+
+	return instance, binding, service, plan, nil
+}
+
+// ComputeExpectedResources computes expected resources
+func ComputeExpectedResources(client kubernetes.Client, instanceID, bindingID, serviceID, planID, action, namespace string) ([]*unstructured.Unstructured, error) {
+	instance, binding, service, plan, err := fetchResources(client, instanceID, bindingID, serviceID, planID, namespace)
+	if err != nil {
+		log.Printf("error getting resource. %v\n", err)
+		return nil, err
+	}
+
+	if plan == nil || service == nil {
+		return nil, fmt.Errorf("failed to get service or plan details")
+	}
+
+	name := types.NamespacedName{
+		Namespace: namespace,
+		Name:      instance.GetName(),
+	}
+
+	switch action {
+	case osbv1alpha1.BindAction:
+		name.Name = binding.GetName()
+	}
+
+	template, err := plan.GetTemplate(action)
+	if err != nil {
+		log.Printf("plan %s does not have %s template. %v\n", planID, action, err)
+		return nil, err
+	}
+
+	renderer, err := rendererFactory.GetRenderer(template.Type, nil)
+	if err != nil {
+		log.Printf("error getting renderer of type %s. %v\n", template.Type, err)
+		return nil, err
+	}
+
+	input, err := rendererFactory.GetRendererInput(template, service, plan, instance, binding, name)
+	if err != nil {
+		log.Printf("error creating renderer input of type %s. %v\n", template.Type, err)
+		return nil, err
+	}
+
+	output, err := renderer.Render(input)
+	if err != nil {
+		log.Printf("error renderering resources for service %s. %v\n", serviceID, err)
+		return nil, err
+	}
+
+	files, err := output.ListFiles()
+	if err != nil {
+		log.Printf("error listing rendered resource files for service %s. %v\n", serviceID, err)
+		return nil, err
+	}
+
+	resources := make([]*unstructured.Unstructured, 0, len(files))
+	for _, file := range files {
+		subResourcesString, err := output.FileContent(file)
+		if err != nil {
+			log.Printf("error getting file content %s. %v\n", file, err)
+			continue
+		}
+
+		subresources, err := dynamic.StringToUnstructured(subResourcesString)
+		if err != nil {
+			log.Printf("error converting file content to unstructured %s. %v\n", file, err)
+			continue
+		}
+
+		for _, obj := range subresources {
+			obj.SetNamespace(namespace)
+			resources = append(resources, obj)
+		}
+	}
+	return resources, nil
+}
+
+// SetOwnerReference updates the owner reference for all the resources
+func SetOwnerReference(owner metav1.Object, resources []*unstructured.Unstructured, scheme *runtime.Scheme) error {
+	for _, obj := range resources {
+		if err := controllerutil.SetControllerReference(owner, obj, scheme); err != nil {
+			log.Printf("error setting owner reference for resource. %v\n", err)
+			continue
+		}
+	}
+	return nil
```

Comment:
Doesn't this function always return nil ?
When is the error returned ? if not we dont need a return parameter

Reply:
Updating it to fail and propagate the error